### PR TITLE
Merge optim tuples

### PR DESCRIPTION
**Pull request by fehiepsi**
Address part of #274

### CODE REVIEW

File: docs/source/mcmc.rst [hunk 1]
```diff
@@ -22,8 +22,6 @@ MCMC Utilities
 
 .. autofunction:: numpyro.util.fori_collect
 
-.. autofunction:: numpyro.diagnostics.summary
```
Comment (fehiepsi) [inline docs/source/mcmc.rst]: this is available in diagnostic docs

File: test/test_svi.py [hunk 2]
```diff
@@ -25,8 +25,8 @@ def guide():
                        constraint=constraints.positive)
         sample("beta", dist.Beta(alpha_q, beta_q))
 
-    opt_init, opt_update, get_params = optimizers.adam(0.05)
-    svi_init, svi_update, _ = svi(model, guide, elbo, opt_init, opt_update, get_params)
+    _, _, get_params = optim = optimizers.adam(0.05)
```
Comment (neerajprad) [inline test/test_svi.py]: I have a suggestion. How about we just wrap the `optimizers` module in jax, and just make these available as class methods? Kind of what we do in Pyro. Then users can just call `opt.get_params()`, or `opt.update()`, etc. That will also help us pass `opt` directly to `svi` without having to do any packing, and pass the same to autoguides. I think the stax and optimizers modules are designed to be used as substrates for more complex functionality anyways.

Comment (neerajprad) [reply test/test_svi.py]: I think this is a good intermediate solution anyways, and am happy to merge this and handle this in a separate PR.

Comment (fehiepsi) [reply test/test_svi.py]: I vote for it. Though currently, it seems not necessary, I think it is potentially useful later on (e.g. if we need to create something similar to ClippedAdam, users won't get confused about where to import this optimizer: from jax.experimental.optimizers or numpyro.optim...).

Comment (fehiepsi) [reply test/test_svi.py]: There are some merge conflicts so I'll try to make a class version for this while addressing them.

**Comment by fehiepsi** [comment]
Let's close this because there are many merge conflicts. It is better to rewrite the change with the new optim class.

**Pull request closed without merging**